jssource/jsmin.php

```diff
@@ -45,4 +45,50 @@ protected function jsParser() {
         require_once('jssource/Minifier.php');
         return Minifier::minify($this->text);
 	}
+
+    /**
+     * Join and minify JS files
+     *
+     * @param array $jsFiles an 'array' of js files
+     *
+     * @author Jose C. Massón <jose@gcoop.coop>
+     *
+     * @return Minified JS file path
+     */
+    static public function joinAndMinifyJSFiles($jsFiles)
+    {
+        $target = SugarThemeRegistry::current()->getJSPath()
+                . '/' .
+                sha1(implode('|', $jsFiles)) . '.js';
+
+        if (!is_file(sugar_cached($target))) {
+            $customJSContents = '';
+
+            foreach ($jsFiles as $jsFileName) {
+                $jsFileContents = '';
+
+                if (is_file($jsFileName)) {
+                    $jsFileContents .= sugar_file_get_contents($jsFileName);
+                }
+
+                if (empty($jsFileContents)) {
+                    LoggerManager::getLogger()->warn(
+                        translate('ERR_FILE_EMPTY') .': '. "{$jsFileName}"
+                    );
+                }
+                $customJSContents .= $jsFileContents;
```

Comment:
value of $jsFileContents could came from sugar_file_get_contents(). In case if function sugar_file_get_contents() failing (for e.g because the file is not readable) then the error results (`false`) just concatenated to $customJSContents but the `false` converted to an empty string just like if there is no concatenation. It means if a file permission problem (or anything) occured we never will realise it. Please handle the error results properly.

Reply:
Done